### PR TITLE
Feat/usage reporting page

### PR DESCRIPTION
**Pull request by johnyrahul**
## What

 Push page usage every time an extraction takes place irrespective of the text extractor used

## Why

Report pages used metric to platform

## How

Pushing the data to platform service similar to what is done for token usage reporting.

## Relevant Docs

-

## Related Issues or PRs

- https://github.com/Zipstack/unstract/pull/547

## Dependencies Versions / Env Variables

-

## Notes on Testing

Tested the SDK with backend
Tested the SDK with structure and text extractor tools

## Screenshots

image
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/sdk/utils/tool_utils.py [hunk 1]
```diff
@@ -100,3 +100,20 @@ def get_file_mime_type(input_file: Path) -> str:
...
+    @staticmethod
+    def get_file_size(input_file: Path) -> int:
+        """Gets the file size in bytes for an input file.
+        Args:
+            input_file (Path): Path object of the input file
+
+        Returns:
+            str: MIME type of the file
+        """
+        with open(input_file, mode="rb") as input_file_obj:
```
Comment (gaya3-zipstack) [inline src/unstract/sdk/utils/tool_utils.py:113]: Should we handle basic errors like file not found here @johnyrahul ?

File: src/unstract/sdk/audit.py [hunk 2]
```diff
@@ -113,3 +113,49 @@ def push_usage_data(
         finally:
             if isinstance(token_counter, TokenCountingHandler):
                 token_counter.reset_counts()
+
+    def push_page_usage_data(
+        self,
+        platform_api_key: str,
```
Comment (gaya3-zipstack) [inline src/unstract/sdk/audit.py:119]: Is it not possible to get platform key also from env @johnyrahul ?

File: src/unstract/sdk/utils/tool_utils.py (hunk 1 above)
Comment (johnyrahul) [reply src/unstract/sdk/utils/tool_utils.py:113]: @gaya3-zipstack The calling function is encapsulated with try catch.
Calling the process from the index function.

**Review by chandrasekharan-zipstack — APPROVED** [review]

File: src/unstract/sdk/utils/tool_utils.py [hunk 3]
```diff
@@ -100,3 +100,20 @@ def get_file_mime_type(input_file: Path) -> str:
...
         return input_file_mime
+
+    @staticmethod
+    def get_file_size(input_file: Path) -> int:
+        """Gets the file size in bytes for an input file.
+        Args:
+            input_file (Path): Path object of the input file
+
+        Returns:
+            str: MIME type of the file
```
Comment (Deepak-Kesavan) [inline src/unstract/sdk/utils/tool_utils.py:111]: @johnyrahul Can you update the docstring here?

File: src/unstract/sdk/x2txt.py [hunk 4]
```diff
@@ -92,3 +109,32 @@ def get_x2text(self, adapter_instance_id: str) -> X2TextAdapter:
...
+        if mime_type == MimeType.PDF:
+            with pdfplumber.open(input_file_path) as pdf:
+                # calculate the number of pages
+                max_pages = len(pdf.pages)
+            Audit().push_page_usage_data(
+                platform_api_key=self._tool.get_env_or_die(ToolEnv.PLATFORM_API_KEY),
+                file_name=file_name,
+                file_size=file_size,
+                file_type=mime_type,
+                page_count=max_pages,
```
Comment (Deepak-Kesavan) [inline src/unstract/sdk/x2txt.py:127]: @johnyrahul What if the pages to extract is explicitly mentioned in the LLM whisperer adapter? Always considering max_pages here won't work.

**Review by Deepak-Kesavan — CHANGES_REQUESTED** [review]

File: src/unstract/sdk/x2txt.py (hunk 4 above)
Comment (johnyrahul) [reply src/unstract/sdk/x2txt.py]: Thanks @Deepak-Kesavan  for pointing it out, Addressed the change and tested.

**Review by Deepak-Kesavan — APPROVED** [review]

**Review by Deepak-Kesavan — APPROVED** [review]

**Review by Deepak-Kesavan — APPROVED** [review]

**Pull request merged**